framework/sandstone.h

```diff
@@ -145,6 +145,42 @@ extern "C" {
 
 #define END_DECLARE_TEST   };
 
+/**
+ * @brief Variadic macro to build NUMBER object on various number of arguments.
+ *
+ * Macro allows to use variadic calls in C language.
+ * Example instantiation:
+ * `#define V(...) OVERLOAD(V,NARGS(__VA_ARGS__))(__VA_ARGS__)`
+ * `#define V1(a) build_float((a))`
+ * `#define V2(a, b) build_rational((a), (b))`
+ *
+ * after expansion V might be used with single (e.g. where float value is passed)
+ * or with two arguments (e.g. where number might be expressed as rational number)
+ * All possible expressions must be defined, otherwise the code won't compile.
+ * @{
+ */
+#define NARGN(\
+            n1, n2, n3, n4, n5, n6, n7, n8, n9, n10,n11,n12,n13,n14,n15,n16,n17,n18,n19,n20,n21,n22,n23,n24,n25,n26,n27,n28,n29,n30,n31,\
+            n32,n33,n34,n35,n36,n37,n38,n39,n40,n41,n42,n43,n44,n45,n46,n47,n48,n49,n50,n51,n52,n53,n54,n55,n56,n57,n58,n59,n60,n61,n62,n63,\
+            N, ...) \
+        N
+
+#define NARG_(...) NARGN(__VA_ARGS__)
+#define NARGS(...) NARG_( \
+            __VA_ARGS__,\
+            63,62,61,60,59,58,57,56,55,54,53,52,51,50,49,48,47,46,45,44,43,42,41,40,39,38,37,36,35,34,33,32,\
+            31,30,29,28,27,26,25,24,23,22,21,20,19,18,17,16,15,14,13,12,11,10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0)
+
+#define OVERLOAD_(name,num) name##num
+#define OVERLOAD(name,num) OVERLOAD_(name,num)
+/** @} */
+
+#define IS_ALIGNED(ptr, a) ((((uint64_t) (ptr)) & ((a) - 1)) == 0)
+
+#define MEMBER_SIZE(type, member) sizeof(((type *)0)->member)
```

Comment:
Not valid for C++, and doubtful it's valid even for C. But for C++ you would never write this, you'd just write `sizeof(type::member)`, so no problem.

Reply:
It was intended to be used in C only (and sandstone.**h** suggest C language, while it is a mix of C/C++).
BTW. Removed, as is not used (was a part of RationalNumber used in matrix_inverse, which is not a part of any commit).